app/controllers/SchemeCacheController.scala

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import org.joda.time.DateTime
+import play.api.libs.json.JodaWrites._
+import play.api.libs.json._
+import play.api.mvc.{Action, AnyContent, ControllerComponents}
+import play.api.{Configuration, Logger}
+import repositories.SchemeCacheRepository
+import uk.gov.hmrc.auth.core.{AuthConnector, AuthorisedFunctions}
+import uk.gov.hmrc.play.bootstrap.controller.BackendController
+
+import scala.concurrent.{ExecutionContext, Future}
+
+abstract class SchemeCacheController(
```

Comment:
Everything in this class is a duplication of PensionSchemeCacheController except for the repository used. Therefore could we not have both cache controllers extending another abstract class on which the methods are defined? That would get rid of the code duplication.